### PR TITLE
3423 incremental updates to support multiple db replications

### PR DESCRIPTION
**Pull request by vimemo**
@SCdF, here is the latest. I'll try to be available when you come back to chat about this. Updated things to avoid so many env variables and using the cli with some defaults instead. We can chat about it later to see how you feel about this :)

### CODE REVIEW

File: libs/medic.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      pgp = require('pg-promise'),
+      Promise = require('es6-promise').Promise,
+      log = require('./log');
+
+// Removes credentials from couchdb url
+// Converts http://admin:pass@localhost:5984/couch1
+// to localhost:5984/couch1 -- seq source
+const parseSource = url => {
+  const source = urlParser.parse(url);
+  return `${source.host}${source.path}`;
```
Comment (SCdF) [inline libs/medic.js]: Nice idea!

File: spec/bootstrap.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+const waitForCouch = async (url) => {
+  await waitForDb({url: url, fn: async () => {
+    try {
+      const {statusCode} = await curl.get(url);
+      if(statusCode === 200) {
+        console.log(`- Couch [${url}] is now avaliable.`);
+        return true;
+      }
+    } catch(err) {
+      //Ignore
```
Comment (SCdF) [inline spec/bootstrap.js]: Why would you want to ignore this?

File: spec/replicate.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  });
+
+  it('replicates', async () => {
+    expect(medicDocs.length + sentinelDocs.length).to.equal(4);
+
+    await medic(COUCH_DB_URL).replicateTo(PG_DB_URL, 1);
+    const pgconn = knex({client: 'pg', connection: PG_DB_URL});
+    let rows = await pgconn.raw('select * from couchdb');
+    expect(rows.rowCount).to.equal(4);//Avoids view
+    await pgconn.destroy();
```
Comment (SCdF) [inline spec/replicate.js]: It doesn't really matter when there is one test, but perhaps this should be in an `afterEach`?

File: libs/runner.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+const run = async (couchUrl, pgconn, timesToRun=undefined) => {
+  log.info('Beginning couch2pg and xmlforms run at ' + new Date());
+  const [allResults, runErrored] = await replicateAll(couchUrl, pgconn);
+  if(allResults) {
+    const [medicResults, sentinelResults] = allResults;
+    // Run secondary tasks if we reasonably think there might be new data
+    // runErrored || errorCount <- something went wrong, but maybe there is still new data
+    // firstRun <- this is first run, we don't know what the DB state is in
+    // results.deleted.length || results.edited.length <- data has changed
```
Comment (SCdF) [inline libs/runner.js]: This comment is now out of date. If you believe it's self-explanatory delete the comment, or if you think it's a useful comment keep it up to date.

File: spec/migrate.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+const assert = require('chai').assert,
```
[the remark this answers is not in the document]
Comment (vimemo) [reply spec/migrate.js]: This repo did not have any tests but was initially setup with a test directory for unit tests and spec for integration tests. I'll update to use just test dir.

File: spec/utils/pgutils.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
+const urlParser = require('url'),
+      knex  = require('knex');
+
+const ensureDbExists = async (url) => {
+  const opts = urlParser.parse(url);
+  const dbname = opts.path.slice(1);
+  const conn = knex({client: 'pg', connection: url.replace(opts.path, '')});
+  try {
+    await conn.raw(`CREATE DATABASE \"${dbname}\"`);
```
[the remark this answers is not in the document]
Comment (vimemo) [reply spec/utils/pgutils.js]: Seems to be necessary for names like medic-analytics; dash.

File: spec/bootstrap.js (hunk 2 above)
Comment (vimemo) [reply spec/bootstrap.js]: good catch; pun intended. need to document or remove this.

File: package.json [hunk 7]
```diff
@@ -1,35 +1,44 @@
 {
   "name": "medic-couch2pg",
-  "version": "1.2.1",
+  "version": "1.3",
```
Comment (SCdF) [inline package.json]: 2.0 plz, you've removed stuff, changed the way the CLI works etc, it definitely is an entirely new deal.

**Pull request merged**